rsc/client/opendota/match.go

```diff
@@ -0,0 +1,58 @@
+package opendota
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	ErrFetchMatchDetail = fmt.Errorf("failed to fetch match detail")
+)
+
+type OpenDotaAPI struct {
+	httpClient *http.Client
+	APIURL     string
+}
+
+func NewMatchAPI(httpClient *http.Client, APIURL string) *OpenDotaAPI {
+	return &OpenDotaAPI{
+		httpClient: httpClient,
+		APIURL:     APIURL,
+	}
+}
+
+func (m *OpenDotaAPI) FetchMatchDetail(ctx context.Context, matchID int64) (MatchDetail, error) {
+	matchDetailURL := fmt.Sprintf("%s/matches/%d", m.APIURL, matchID)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, matchDetailURL, nil)
+	if err != nil {
+		return MatchDetail{}, err
+	}
+
+	res, err := m.httpClient.Do(req)
+	if err != nil {
+		return MatchDetail{}, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return MatchDetail{}, ErrFetchMatchDetail
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return MatchDetail{}, err
+	}
+
+	var matchDetail MatchDetail
```

Comment:
maybe declare at the beginning and you can use `return matchDetail, error` for error responses. or perhaps you think `return MatchDetails{}, error` makes it clearer that there is an error?

Reply:
> or perhaps you think return MatchDetails{}, error makes it clearer that there is an error?

Yup, I think like that. Sometimes people might question why the code returns a variable that has no value. Therefore, I differentiate between returning a value and returning no value by using  `MatchDetails{}`). What do you think?

Reply:
sounds reasonable, we can leave it this way then.